controllers/apply-next/helpers.js

```diff
@@ -1,186 +1,268 @@
+ * Or, try common display formats based on type
+ * Otherwise, call toString on the result
+ *
+ * @param {any} value
+ * @param {object} field
+ */
+function toDisplayValue(value, field) {
+    if (field.displayFormat) {
+        return field.displayFormat.call(field, value);
+    } else if (field.type === 'radio') {
+        const optionMatch = find(field.options, option => option.value === value);
+        return optionMatch ? optionMatch.label : value.toString();
+    } else if (field.type === 'date') {
+        return moment(value).format('D MMMM, YYYY');
+    } else {
+        return value.toString();
+    }
+}
+
+/**
+ * Enhances a form object by:
+ * - Localising all labels and messages
+ * - Assigning values to fields, along with a display value for views
+ * - Marking steps as notRequired if matchesCondition is currently false
+ *
+ * @param {String} locale
+ * @param {Object} form
+ * @param {Object} data
+ */
+function enhanceForm(locale, form, data) {
     const localise = get(locale);
+    const clonedForm = cloneDeep(form);
 
-    const translateSection = section => {
+    const enhanceSection = section => {
         section.title = localise(section.title);
         if (section.introduction) {
             section.introduction = localise(section.introduction);
         }
         return section;
     };
 
-    const translateField = field => {
+    const enhanceField = field => {
         field.label = localise(field.label);
         field.explanation = localise(field.explanation);
-        const match = find(formData, (value, name) => name === field.name);
-        if (match) {
-            field.value = match;
-        }
 
-        // Translate each option (if set)
         if (field.options) {
             field.options = field.options.map(option => {
                 option.label = localise(option.label);
                 option.explanation = localise(option.explanation);
                 return option;
             });
         }
+
+        // Assign value to field if present
+        const fieldValue = find(data, (value, name) => name === field.name);
+        if (fieldValue) {
+            field.value = fieldValue;
+            field.displayValue = toDisplayValue(fieldValue, field);
+        }
+
         return field;
     };
 
-    const translateStep = step => {
+    const enhanceStep = step => {
         step.title = localise(step.title);
 
         step.fieldsets = step.fieldsets.map(fieldset => {
             fieldset.legend = localise(fieldset.legend);
             fieldset.introduction = localise(fieldset.introduction);
-            fieldset.fields = fieldset.fields.map(translateField);
+            fieldset.fields = fieldset.fields.map(enhanceField);
             return fieldset;
         });
 
+        // Handle steps that don't need to be completed based on current form data
+        if (step.matchesCondition && step.matchesCondition(data) === false) {
+            step.notRequired = true;
```

Comment:
Moved the `notRequired` step up into this main function as it lets us simplify the "completion status" function we use on the summary page and means we can perform this check earlier.